File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        long ln = Long.valueOf(num);
        
        if(ln == 1) return 0;
        
        while(ln != 1) {
            if(ln % 2 == 0) {
                ln /= 2;
                answer++;
            }
            else {
                ln *= 3;
                ln++;
                answer++;
            }
            if(answer == 500) return -1;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    if num == 1:
        return 0

    return collatzation(num, 0)

def collatzation(num, count):
    if num == 1:
        return count
    
    if count == 500:
        return -1
    
    if num % 2 == 0:
        num = int(num / 2)
    else:
        num *= 3
        num += 1

    return collatzation(num, count + 1)